classes/Address.php

```diff
@@ -232,6 +236,11 @@ public function delete()
         if (!$this->isUsed()) {
             $this->deleteCartAddress();
 
+            // Update the cache
+            if (isset(static::$addressExists[$this->id])) {
```

Comment:
You could also update this in the `add` and `update` methods if you want to optimize a little bit more